### PR TITLE
refactor(nakama): break cardinal request function to reduce complexity

### PR DESCRIPTION
**Pull request by namrapatel**
Closes: WORLD-743

## Brief Changelog

- Break `handleCardinalRequest` into a sub-function to get rid of the `nolint:gocognit` usage

## Testing and Verifying

- This change is a trivial rework/code cleanup without any test coverage.

### CODE REVIEW

File: relay/nakama/cardinal.go [hunk 1]
```diff
@@ -40,3 +44,50 @@ func getCardinalEndpoints() (txEndpoints []string, queryEndpoints []string, err
...
+	req, err := http.NewRequestWithContext(
+		ctx,
+		http.MethodPost,
+		utils.MakeHTTPURL(endpoint, globalCardinalAddress),
+		payload,
+	)
+	req.Header.Set("Content-Type", "application/json")
+	if err != nil {
+		return res, eris.Wrapf(err, "request setup failed for endpoint %q", endpoint)
+	}
```
Comment (technicallyty) [inline relay/nakama/cardinal.go:63]: move this up to where the err is created

File: relay/nakama/cardinal.go [hunk 2]
```diff
@@ -40,3 +44,50 @@ func getCardinalEndpoints() (txEndpoints []string, queryEndpoints []string, err
...
+	if err != nil {
+		return res, eris.Wrapf(err, "request failed for endpoint %q", endpoint)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return res, eris.Wrapf(err, "failed to read response body, bad status code: %s: %s", resp.Status, body)
+		}
+		return res, eris.Errorf("bad status code: %d: %s", resp.StatusCode, body)
```
Comment (technicallyty) [inline relay/nakama/cardinal.go:74]: do we wanna use `StatusCode` or `Status`?

**Review by jerargus — APPROVED** [review]

File: relay/nakama/cardinal.go [hunk 3]
```diff
@@ -40,3 +44,50 @@ func getCardinalEndpoints() (txEndpoints []string, queryEndpoints []string, err
...
+	}
+	bz, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return res, eris.Wrap(err, "can't ready body")
```
Comment (technicallyty) [inline relay/nakama/cardinal.go:78]: you were more descriptive in the error returned in the first call to `ReadAll`, I prefer that error to this one

File: relay/nakama/cardinal.go [hunk 4]
```diff
@@ -40,3 +44,50 @@ func getCardinalEndpoints() (txEndpoints []string, queryEndpoints []string, err
...
+	}
+	if strings.HasPrefix(endpoint, TransactionEndpointPrefix) {
+		var asTx persona.TxResponse
+
+		if err = json.Unmarshal(bz, &asTx); err != nil {
+			return res, eris.Wrap(err, "can't decode body as tx response")
```
Comment (technicallyty) [inline relay/nakama/cardinal.go:84]: prefer saying `failed to decode body`

**Pull request merged**